src/conn/conn_layered.c

```diff
@@ -2110,11 +2113,13 @@ __layered_drain_ingest_tables(WT_SESSION_IMPL *session)
     if (multithreaded) {
         __wt_cond_signal(session, conn->layered_drain_data.threads.wait_cond);
         __wt_writelock(session, &conn->layered_drain_data.threads.lock);
-        WT_TRET(__wt_thread_group_destroy(session, &conn->layered_drain_data.threads));
+        if (group_created)
+            WT_TRET(__wt_thread_group_destroy(session, &conn->layered_drain_data.threads));
     }
     /* Cleanup and release resources. */
     __layered_drain_clear_work_queue(session);
-    WT_TRET(__wt_session_close_internal(internal_session));
+    if (internal_session != NULL)
```

Comment:
Should internal_session be initialised to NULL?

Reply:
Yes, would be better: 92611dc386616c770dc065ed0515a5bc86bfa457